### ISSUE TITLE
Remove (or disable) Style/BracesAroundHashParameters

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#7641]
Since Ruby 2.7, the existence of curly braces on the last argument has meaning.

It means, the following codes have different meanings.

```ruby
foo(x: 1) # x: 1 is a keyword argument
foo({x: 1}) # {x: 1} is a hash
```

There are the correct pairs of method definitions and method calls.
```ruby
def foo(x: 1)
end
foo(x: 1)

def foo(hash)
end
foo({x: 1})
```

For more detail https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

So I propose removing the cop. I think we should disable the cop by default at least.

--------

## Expected behavior

Remove or disable the cop.

## Actual behavior

It's available.

## Steps to reproduce the problem

nothing

## RuboCop version

master

**Comment by koic** [comment]
I agree with @pocke. I think this proposal makes sense for future Ruby 3. I think the cop should be removed to smooth the transition from Ruby 2 to Ruby 3.

**Comment by bbatsov** [comment]
Yeah, agreed. Let's remove it.

**Comment by deivid-rodriguez** [comment]
Hi @pocke!

Currently the `HashSyntax` cop also warns `foo(x: 1)` invocations if configured with the hash rockets style. I'm guessing these offences should also be removed from the `Style/HashSyntax` cop, right?

**Comment by sandstrom** [comment]
Agree with what @deivid-rodriguez said, it would be awesome if `Style/HashSyntax` configured with `hash_rockets` wouldn't complain about keyword arguments, e.g. `foo(x: 1)`.

Also, see https://github.com/rubocop-hq/rubocop/issues/1451
